client/cli/src/client_proxy.rs

```diff
@@ -148,7 +150,9 @@ impl ClientProxy {
 
         let faucet_server = match faucet_server {
             Some(server) => server,
-            None => host
+            None => url
+                .host_str()
+                .ok_or_else(|| format_err!("Missing host in URL"))?
                 .replace("http://", "")
                 .replace("https://", "")
                 .replace("client", "faucet"),
```

Comment:
Does this normally appear in the host position?

Reply:
yes - for this branch we expect a very specific URL